dartsim/src/SimulationFeatures.cc

```diff
@@ -52,7 +59,54 @@ void SimulationFeatures::WorldForwardStep(
 
   // TODO(MXG): Parse input
   world->step();
-  // TODO(MXG): Fill in output and state
+  this->WriteRequiredData(_h);
+  this->Write(_h.Get<JointPositions>());
+  // TODO(MXG): Fill in state
+}
+
+void SimulationFeatures::Write(JointPositions &/*_positions*/) const
+{
+  // TODO(adlarkin) implement this, if it's needed?
+}
+
+void SimulationFeatures::Write(WorldPoses &_poses) const
+{
+  // cache link poses from the previous iteration
+  // (using a static variable for now instead of a member variable
+  // since this method has to be const)
+  static std::unordered_map<std::size_t, math::Pose3d> prevLinkPoses;
```

Comment:
As mentioned in the comment, I couldn't use a member variable since the `Write` override method is required to be `const` (so, since this method is `const`, I can't modify member variables inside of it). Any thoughts on how to get around this other than using a static variable?

Reply:
you could mark `prevLinkPoses` as `mutable` if you need to modify it in a const function

Reply:
Thanks, I wasn't aware of the `mutable` keyword - I learned something new! This has been added in https://github.com/ignitionrobotics/ign-physics/pull/223/commits/baeb7e6606c21ade93ced6af5d96c6139f41d878